app/src/main/java/org/schabi/newpipe/util/StreamDialogEntry.java

```diff
@@ -203,4 +210,39 @@ private static void openChannelFragment(final Fragment fragment,
                 fragment.requireActivity().getSupportFragmentManager(),
                 item.getServiceId(), uploaderUrl, item.getUploaderName());
     }
+
+    /////////////////////////////////////////////
+    // helper functions                        //
+    /////////////////////////////////////////////
+
+    private static void fetchItemInfoIfSparse(final Fragment fragment,
+            final StreamInfoItem item,
+            final Consumer<SinglePlayQueue> callback) {
```

Comment:
What is the advantage of using a consumer for a `SinglePlayQueue`? The method is more generic if the consumer accepts a `StreamInfoItem`.

Reply:
This was because getStreamInfo returns a `StreamInfo` not a `StreamInfoItem`, forgive me if this exists but I didn't find a way to easily cast/convert between these types.

Reply:
Yes, this is ok